native_client/deepspeech.h

```diff
@@ -4,83 +4,119 @@
-     *
-     * @return A DeepSpeech context.
-     */
-    DeepSpeech(const char* aModelPath, int aNCep, int aNContext);
-    ~DeepSpeech();
+  class Model {
+    private:
+      Private* mPriv;
 
-    /**
-     * @brief Extract MFCC features from a given audio signal and add context.
-     *
-     * Extracts MFCC features from a given audio signal and adds the appropriate
-     * amount of context to run inference with the given DeepSpeech context.
-     *
-     * @param aBuffer A 16-bit, mono raw audio signal at the appropriate sample
-     *                rate.
-     * @param aBufferSize The sample-length of the audio signal.
-     * @param aSampleRate The sample-rate of the audio signal.
-     * @param[out] aMFCC An array containing features, of shape
-     *                   (@p aNFrames, ncep * ncontext). The user is responsible
-     *                   for freeing the array.
-     * @param[out] aNFrames (optional) The number of frames in @p aMFCC.
-     * @param[out] aFrameLen (optional) The length of each frame
-     *                       (ncep * ncontext) in @p aMFCC.
-     */
-    void getMfccFrames(const short* aBuffer,
-                       unsigned int aBufferSize,
-                       int aSampleRate,
-                       float** aMfcc,
-                       int* aNFrames = NULL,
-                       int* aFrameLen = NULL);
+    public:
+      /**
+       * @brief Initialise a DeepSpeech context.
+       *
+       * @param aModelPath The path to the frozen model graph.
+       * @param aNCep The number of cepstrum the model was trained with.
+       * @param aNContext The context window the model was trained with.
+       *
+       * @return A DeepSpeech model.
+       */
+      Model(const char* aModelPath, int aNCep, int aNContext);
+      ~Model();
 
-    /**
-     * @brief Run inference on the given audio.
-     *
-     * Runs inference on the given MFCC audio features with the given DeepSpeech
-     * context. See DsGetMfccFrames().
-     *
-     * @param aMfcc MFCC features with the appropriate amount of context per
-     *              frame.
-     * @param aNFrames The number of frames in @p aMfcc.
-     * @param aFrameLen (optional) The length of each frame in @p aMfcc. If
-     *                  specified, this will be used to verify the array is
-     *                  large enough.
-     *
-     * @return The resulting string after running inference. The user is
-     *         responsible for freeing this string.
-     */
-    char* infer(float* aMfcc,
-                int aNFrames,
-                int aFrameLen = 0);
+      /**
+       * @brief Given audio, return a vector suitable for input to the
+       *        DeepSpeech model.
+       *
+       * Extracts MFCC features from a given audio signal and adds the
+       * appropriate amount of context to run inference on the DeepSpeech model.
+       *
+       * @param aBuffer A 16-bit, mono raw audio signal at the appropriate
```

Comment:
Would it make sense we state that "appropriate" is what the model has been trained against ?